tests/testthat/test-Logger.R

```diff
@@ -1,240 +1,475 @@
+    expect_length(db_log_file, 0)
+
+    # Clean up
     connection_clean_up(conn)
+    rm(logger)
+    invisible(gc())
   }
 })
 
-test_that("Logger$finalize handles log table is at some point deleted", {
+
+test_that("Logger: $finalize() handles log table is at some point deleted", {
   for (conn in get_test_conns()) {
 
+    # Set options for the test
+    db_table <- "test.SCDB_logger"
+    timestamp <- "2022-06-01 09:00:00"
+
     log_table_id <- "expendable_log_table"
-    logger <- Logger$new(log_conn = conn,
-                         log_table_id = log_table_id,
-                         log_path = NULL)
+    logger <- Logger$new(db_table = db_table, timestamp = timestamp, log_conn = conn, log_table_id = log_table_id)
 
-    DBI::dbRemoveTable(conn, log_table_id)
+    DBI::dbRemoveTable(conn, id(log_table_id, conn))
 
     expect_no_error(logger$finalize())
-    gc()
 
+    # Clean up
     connection_clean_up(conn)
+    rm(logger)
+    invisible(gc())
   }
 })
+
+test_that("Logger: custom timestamp_format works", {
+
+  # Create logger and test configuration
+  expect_warning(
+    logger <- Logger$new(),
+    regexp = "NO file or DB logging will be done."
+  )
+
+  # Test logging to console has the right formatting and message type
+  ts_str <- format(logger$start_time, "%F %R:%OS3")
+  expect_message(
+    logger$log_info("test console", tic = logger$start_time),
+    glue::glue("{ts_str} - {Sys.info()[['user']]} - INFO - test console")
+  )
+
+  ts_str <- format(logger$start_time, "%F %R")
+  expect_message(
+    logger$log_info("test console", tic = logger$start_time, timestamp_format = "%F %R"),
+    glue::glue("{ts_str} - {Sys.info()[['user']]} - INFO - test console")
+  )
+
+  ts_str <- format(logger$start_time, "%F")
+  withr::local_options("SCDB.log_timestamp_format" = "%F")
+  expect_message(
+    logger$log_info("test console", tic = logger$start_time),
+    glue::glue("{ts_str} - {Sys.info()[['user']]} - INFO - test console")
+  )
+
+  rm(logger)
+  invisible(gc())
+})
+
+
+test_that("LoggerNull: no logging occurs", {
+
+  # Create logger and test configuration
+  expect_no_message(logger <- LoggerNull$new())
+
+  ts_str <- format(logger$start_time, "%F %R:%OS3")
+  expect_no_message(logger$log_info("test console", tic = logger$start_time))
+
+  # Test logging to console has the right formatting and message type
+  ts_str <- format(logger$start_time, "%F %R:%OS3")
+  expect_no_message(
+    logger$log_info("test console", tic = logger$start_time),
+  )
+  expect_warning(
+    logger$log_warn("test console", tic = logger$start_time),
+    glue::glue("{ts_str} - {Sys.info()[['user']]} - WARNING - test console")
+  )
+  expect_error(
+    logger$log_error("test console", tic = logger$start_time),
+    glue::glue("{ts_str} - {Sys.info()[['user']]} - ERROR - test console")
+  )
+
+  expect_no_message(logger$log_to_db(n_insertions = 42))
+  expect_no_message(logger$finalize_db_entry())
+
```

Comment:
Consider adding a test that it does not write to file if getOption("SCDB.log_path") is set.